scalajslib/api/src/ScalaJSWorkerApi.scala

```diff
@@ -36,6 +51,33 @@ object ModuleKind {
   object ESModule extends ModuleKind
 }
 
+sealed trait ModuleSplitStyle
+object ModuleSplitStyle {
+  object FewestModules extends ModuleSplitStyle
+  object SmallestModules extends ModuleSplitStyle
+}
+
+object ModuleInitializer {
+  import upickle.default.{ReadWriter => RW, macroRW}
+  implicit def rw: RW[ModuleInitializer] = macroRW
+}
+
+case class ModuleInitializer(className: String, mainMethod: String, moduleId: String)
+
+sealed trait OutputPatterns
+
+object OutputPatterns {
+  import upickle.default.{ReadWriter => RW, macroRW}
+  implicit def rwFromJsFile: RW[OutputPatternsFromJsFile] = macroRW
+  implicit def rw: RW[OutputPatterns] = macroRW
+
+  def fromJSFile(jsFile: String): OutputPatterns = OutputPatternsFromJsFile(jsFile)
```

Comment:
Since this is internal API, we can probably use the case class apply directly and avoid defining helpers.